bee-message/src/parents.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! The parents module defines the core data type for storing the messages directly approved by a message.
+
+use crate::{MessageId, MessagePackError, MessageUnpackError, ValidationError, MESSAGE_ID_LENGTH};
+
+use bee_ord::is_unique_sorted;
+use bee_packable::{PackError, Packable, Packer, UnpackError, Unpacker};
+
+use bitvec::prelude::*;
+
+use core::ops::{Deref, RangeInclusive};
+
+use alloc::{vec, vec::Vec};
+
+/// The range representing the valid number of parents.
+pub const MESSAGE_PARENTS_RANGE: RangeInclusive<usize> = 1..=8;
+
+/// Minimum number of strong parents for a valid message.
+pub const MESSAGE_MIN_STRONG_PARENTS: usize = 1;
+
+/// An individual message parent, which can be categorized as "strong" or "weak".
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(tag = "type", content = "data")
+)]
+pub enum Parent {
+    /// Message parents in which the past cone is "Liked".
+    Strong(MessageId),
+    /// Message parents in which the past cone is "Disliked", but the parents themselves are "Liked".
+    Weak(MessageId),
+}
+
+impl Parent {
+    /// Returns the `MessageId` of this `Parent`.
+    pub fn id(&self) -> &MessageId {
+        match self {
+            Self::Strong(id) => id,
+            Self::Weak(id) => id,
+        }
+    }
+}
+
+/// A [`Message`]'s `Parents` are the [`MessageId`]s of the messages it directly approves.
+///
+/// Parents must be:
+/// * in the `MESSAGE_PARENTS_RANGE` range;
+/// * lexicographically sorted;
+/// * unique;
```

Comment:
Add something about `MESSAGE_MIN_STRONG_PARENTS` ?

Reply:
Sure -- I'll also add similar comments to other validated types.